### ISSUE TITLE
Exceptions when running codecarbon from a thread

### ISSUE THREAD

**Issue by ivanignatiev** [mlco2/codecarbon#732]
* CodeCarbon version: 2.8.0
* Python version: 3.12
* Operating System: Ubuntu

### Description

When codecarbon OfflineEmissionsTracker is running from a thread there are some exceptions : 

```python
codecarbon_tracker = OfflineEmissionsTracker(country_iso_code="FRA", save_to_file=False)
```

```
== APP == [codecarbon INFO @ 20:02:49] offline tracker init
== APP == [codecarbon WARNING @ 20:02:49] graceful shutdown. Exceptions:
== APP == [codecarbon WARNING @ 20:02:49] <class 'Exception'>
== APP == Traceback (most recent call last):
== APP ==   File "/home/ignatiev/projects/****/.venv/lib/python3.12/site-packages/codecarbon/core/util.py", line 24, in suppress
== APP ==     yield
== APP ==   File "/usr/lib/python3.12/contextlib.py", line 81, in inner
== APP ==     return func(*args, **kwds)
== APP ==            ^^^^^^^^^^^^^^^^^^^
== APP ==   File "/home/ignatiev/projects/****/.venv/lib/python3.12/site-packages/codecarbon/emissions_tracker.py", line 976, in __init__
== APP ==     super().__init__(*args, **kwargs)
== APP ==   File "/home/ignatiev/projects/****/.venv/lib/python3.12/site-packages/codecarbon/emissions_tracker.py", line 251, in __init__
== APP ==     self._lock = Lock()
== APP ==                  ^^^^^^
== APP ==   File "/home/ignatiev/projects/****/.venv/lib/python3.12/site-packages/codecarbon/lock.py", line 29, in __init__
== APP ==     signal.signal(signal.SIGINT, self._handle_exit)  # Ctrl+C
== APP ==     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
== APP ==   File "/usr/lib/python3.12/signal.py", line 58, in signal
== APP ==     handler = _signal.signal(_enum_to_int(signalnum), _enum_to_int(handler))
== APP ==               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
== APP == ValueError: signal only works in main thread of the main interpreter
== APP == [codecarbon WARNING @ 20:02:49] stopping.
== APP == [codecarbon WARNING @ 20:02:49] graceful shutdown. Exceptions:
== APP == [codecarbon WARNING @ 20:02:49] <class 'Exception'>
== APP == Traceback (most recent call last):
== APP ==   File "/home/ignatiev/projects/****/.venv/lib/python3.12/site-packages/codecarbon/core/util.py", line 24, in suppress
== APP ==     yield
== APP ==   File "/usr/lib/python3.12/contextlib.py", line 81, in inner
== APP ==     return func(*args, **kwds)
== APP ==            ^^^^^^^^^^^^^^^^^^^
== APP ==   File "/home/ignatiev/projects/***/.venv/lib/python3.12/site-packages/codecarbon/emissions_tracker.py", line 541, in start
== APP ==     if self._start_time is not None:
== APP ==        ^^^^^^^^^^^^^^^^
== APP == AttributeError: 'OfflineEmissionsTracker' object has no attribute '_start_time'. Did you mean: 'start_task'?
== APP == [codecarbon WARNING @ 20:02:49] stopping.

```

### What I Did

Adding `allow_multiple_runs=True` resolved a problem but I suppose allow_multiple_runs=True is more a workaround than a good practice to handle this issue.

Should be other lock types used?

**Comment by benoit-cty** [comment]
Thanks, signal is used to catch the Ctrl-C for deleting the lock file we use to prevent multiple run.

I think we could just catch and ignore this exception. And just log a warning. @inimaz have you a better idea ?

**Comment by inimaz** [comment]
Thanks! Sadly apparently `signal`  works only in single-thread. We can just warn as you suggest @benoit-cty ! On the long-term, we should as well do a version of this lock multi-thread safe. Otherwise multithread might not remove the lock when they do Ctrl+C.

**Comment by inimaz** [comment]
@ivanignatiev I have added a fix  here --> #737. Note that still If you run codecarbon on 10 threads at the same time (10 instances of codecarbon), only one instance of codecarbon will be really monitoring (since they are sharing CPU, RAM, GPU...) and the other 9 will say 
`Error: Another instance of codecarbon is probably running as we find /tmp/.codecarbon.lock. Turn off the other instance to be able to run this one or use allow_multiple_runs or delete the file. Exiting.`
Is this what you expect?

**Comment by benoit-cty** [comment]
Hello, it has been published in the v2.8.2 of CodeCarbon.
